tests/test_gpg.py

```diff
@@ -809,5 +817,30 @@ def test_verify_short_signature(self):
     self.assertTrue(verify_signature(signature, key, test_data))
 
 
+class TestVersion(unittest.TestCase):
+  """Tests for the Version utility class."""
+
+  def test_version_roundtrip_string(self):
+    """Version parses and formats strings correctly."""
+    for value, expected in [
+        ('1.3.0', Version(1, 3, 0)),
+        ('1.3.1', Version(1, 3, 1)),
+        ('1.3.22', Version(1, 3, 22)),
+    ]:
+      self.assertEqual(Version.from_string(value), expected)
+      self.assertEqual(str(expected), value)
+
+  def test_version_from_string_invalid(self):
+    """Version.from_string rejects invalid inputs."""
+    for value in [
+        '1.3',
+        '1.33.0',
```

Comment:
These are both valid for StrictVersion.

Obviously we don't need to re-implement StrictVersion here (we know this is a hack, it just has to work well enough for this specific case) but at least the latter one seems strange to me -- is this just because gpg historically hasn't used multidigit minor versions?

Reply:
Good question. Unfortunately, I can't remember if it was just an assumption based on the existing version progression (e.g. last 1.x  was 1.9), or if there was a stronger indication somewhere. Happy to change, although not necessarily in this PR, which just uses the regex that was already there.

Reply:
oh you're right: no new bugs then :)